readthedocs_ext/readthedocs.py

```diff
@@ -114,74 +113,84 @@ def rtd_render(self, template, render_context):
                                                         app.builder.templates)
 
 
-def copy_media(app, exception):
-    """ Move our dynamically generated files after build. """
-    if app.builder.name in ['readthedocs', 'readthedocsdirhtml'] and not exception:
-        for file in ['readthedocs-dynamic-include.js_t', 'readthedocs-data.js_t',
-                     'searchtools.js_t']:
-            app.info(bold('Copying %s... ' % file), nonl=True)
-            dest_dir = os.path.join(app.builder.outdir, '_static')
-            source = os.path.join(
-                os.path.abspath(os.path.dirname(__file__)),
-                '_static',
-                file
-            )
-            try:
-                ctx = app.builder.globalcontext
-            except AttributeError:
-                ctx = {}
-            if app.builder.indexer is not None:
-                ctx.update(app.builder.indexer.context_for_searchtool())
-            copy_static_entry(source, dest_dir, app.builder, ctx)
-            app.info('done')
-
-    if 'comments' in app.builder.name and not exception:
-        for file in STATIC_FILES:
-            app.info(bold('Copying %s... ' % file), nonl=True)
-            dest_dir = os.path.join(app.builder.outdir, '_static')
-            source = os.path.join(
-                os.path.abspath(os.path.dirname(__file__)),
-                '_static',
-                file
-            )
-            try:
-                ctx = app.builder.globalcontext
-            except AttributeError:
-                ctx = {}
-            ctx['websupport2_base_url'] = app.builder.config.websupport2_base_url
-            ctx['websupport2_static_url'] = app.builder.config.websupport2_static_url
-            copy_static_entry(source, dest_dir, app.builder, ctx)
-            app.info('done')
+class HtmlBuilderMixin(BuilderMixin):
 
+    static_override_files = [
+        'readthedocs-dynamic-include.js_t',
+        'readthedocs-data.js_t',
```

Comment:
Yeah, Sphinx builders already have `copy_static_files` and a `copy_extra_files`, so these are overrides in that they'll clobber anything set up by `copy_static_files`. I'll think about this a bit more